js/directives.js

```diff
@@ -4,434 +4,760 @@
+                                .scale(x_Anat)
+                                .orient("bottom");
+
+                            var yAxis_Anat = d3.svg.axis()
+                                .scale(y_Anat)
+                                .orient("left")
+                                .ticks(10);
+
+                            var tip_Anat = d3.tip()
+                                .attr('class', 'd3-tip')
+                                .offset([-10, 0])
+                                .html(function(d) {
+                                    return d.key + "<br/>" + "Annotated Taxa Count: " + d3.values(d)[1][0];
+                                })
+
+                            var svg1 = d3.select("#anatomy").append("svg")
+                                .attr("width", width + margin.left + margin.right)
+                                .attr("height", height + margin.top + margin.bottom)
+                                .append("g")
+                                .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+                            svg1.call(tip_Anat);
+
+                            svg1.append("text")
+                                .attr("class", "x label")
+                                .attr("text-anchor", "end")
+                                .attr("x", width / 2)
+                                .attr("y", height + 30)
+
+
+                            svg1.append("g")
+                                .attr("class", "x axis")
+                                .attr("transform", "translate(0," + height + ")")
+                                .call(xAxis_Anat);
+
+                            svg1.selectAll("text")
+                                //.call(wrap, x_Anat.rangeBand())
+                                .attr("y", 10)
+                                .attr("x", 50)
+                                .attr("dx", "-2.8em")
+                                .attr("dy", ".70em")
+                                .attr("transform", "rotate(45)")
+                                .style("text-anchor", "start");
+
+                            //hyperlink the x axis labels
+                            d3.selectAll("text")
+                                .filter(function(d) {
+                                    return typeof(d) == "string";
+                                })
+                                .style("cursor", "pointer")
+                                .on("mouseover", function(d) {
+                                    d3.select(this).style("fill", "blue");
+                                })
+                                .on("mouseout", function(d) {
+                                    d3.select(this).style("fill", "black");
+                                })
+                                .on("click", function(d) {
+                                    document.location.href = "http://kb.phenoscape.org/#/entity/" + data[d][1];
```

Comment:
This line causes a JavaScript error when you click on a taxon name in the X axis of the taxa graph. If you first drill down one level, it works without error.